Volume/View Models/HomeViewModel.swift

```diff
@@ -0,0 +1,216 @@
+//
+//  HomeViewModel.swift
+//  Volume
+//
+//  Created by Hanzheng Li on 10/26/22.
+//  Copyright © 2022 Cornell AppDev. All rights reserved.
+//
+
+import Combine
+import SwiftUI
+
+extension HomeView {
+    @MainActor class ViewModel: ObservableObject {
+        typealias ArticlesResultPublisher = Publishers.Map<OperationPublisher<GetArticlesByPublicationSlugsQuery.Data>, [ArticleFields]>
+
+        private struct Constants {
+            static let pageSize: Double = 10
+        }
+
+        var networkState: NetworkState?
+        var userData: UserData?
+
+        @Published var trendingArticles: MainView.TabState<[Article]> = .loading
+        @Published var weeklyDebrief: MainView.TabState<WeeklyDebrief?> = .loading
+        @Published var followedArticles: MainView.TabState<[Article]> = .loading
+        @Published var unfollowedArticles: MainView.TabState<[Article]> = .loading
+        @Published var hasMoreFollowedArticlePages = true
+        @Published var hasMoreUnfollowedArticlePages = true
+        @Published var isWeeklyDebriefOpen: Bool = false
+        @Published var deeplinkID: String? = nil
+        @Published var openArticleFromDeeplink: Bool = false
+
+        private var publicationSlugs: MainView.TabState<[String]> = .loading
+        private var queryBag = Set<AnyCancellable>()
+
+        /// Returns empty list if still loading
+        private var loadedPublicationSlugs: [String] {
+            switch publicationSlugs {
+            case .loading:
+                return []
+            case .results(let slugs), .reloading(let slugs):
+                return slugs
+            }
+        }
+
+        func setupEnvironment(networkState: NetworkState, userData: UserData) {
+            self.networkState = networkState
+            self.userData = userData
+        }
+
+        var followedPublicationSlugs: [String] {
```

Comment:
Can this be optional instead of an empty array?

Reply:
Intentionally left it non-nil because all places where the nil is handled treats the response like an empty list anyway.